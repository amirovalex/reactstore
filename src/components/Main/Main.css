/*work*/
html {
	margin: 0;
	padding:0;
	position:relative;
	height: 100vh;
}


body {
	height:100%;
	margin: 0;
}

.container {
	padding-top: 5px;
	display: grid;
	grid-template-columns: 1fr 1fr;
	grid-auto-rows: 1fr 1fr;
	grid-auto-flow:column;
	height: 100vh;
	overflow: visible;
	justify-content: center;
    align-content: center;
    filter: drop-shadow(0px 10px 5px rgba(0,0,0,0.5)) drop-shadow(1px 1px 1px #66E7E7) drop-shadow(2px 3px 5px #FFFA49) drop-shadow(2px 1px 10px #C70013);
}

.zone {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
	cursor:pointer;
	font-size: 10em;
	transition: all 0.3s linear;
	background-size: auto;
    background-repeat: no-repeat;
    background-position: center center;
    background-size:cover;
    box-shadow: 5px 5px 3px 5px #ccc;
}

.zone:hover {
	transform:scale(1.02,1.02);
}

.left {
	display:flex;
	align-items: center;
	width:150%;
	position: relative;
	clip-path: polygon(79% 52%,-32% -81%, -56% 221% , 61% 70% , 79% 64% , 84% 61%);
	grid-column: 1;
	grid-row: 1/span 2;
	background-image: url("http://images2.fanpop.com/images/photos/6800000/Slipknot-poster-metal-gods-6809607-2560-1700.jpg");
}

.righttop {
	display:flex;
	align-items: center;
	background-image: url("https://i.ytimg.com/vi/Co1HIvM5KB0/maxresdefault.jpg");
	position: relative;
	width: 150%;
	grid-row: 1;
	clip-path: polygon(4% 0%, 100% 0, 100% 100%, 46% 100%, 38% 82%, 31% 64%, 26% 51%, 16% 28%);
	justify-self: end;
}

.rightlow {
	display:flex;
	align-items: center;
	clip-path: polygon(48% 2%, 100% 2%, 100% 100%, 7% 100%, 14% 81%, 19% 70%, 25% 51% , 41% 34%,52% 24%);
	position: relative;
	width: 150%;
	grid-row: 2;
	justify-self:end;
	background-image: url('http://wallsdesk.com/wp-content/uploads/2016/12/Slipknot-Pictures.jpg');
}

.texter {
	font-size: 4rem;
	font-family: 'Grenze', serif;
}


@media only screen and (max-width: 800px) {
	.texter {
		font-size:2rem;
	}
}
