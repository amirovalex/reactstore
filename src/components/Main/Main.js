import React from 'react';
import './Main.css';
import 'tachyons';

const Main = (props) => {
	console.log('props',props)
	return(
	<div className="container">
		<div 
			onClick={() => {props.onRouteChange('store'); props.onCategoryChange('accessories');props.history.push('/store/accessories')}}
			className="left shader zone tc"><p className='glow texter shadow-1'>Accessories</p></div>
		<div 
			onClick={() => {props.onRouteChange('store'); props.onCategoryChange('topwear');props.history.push('/store/topwear')}}
			className="righttop zone shadow-1"><p className='texter shadow-1'>Topwear</p></div>
		<div 
			onClick={() => {props.onRouteChange('store'); props.onCategoryChange('bottomwear');props.history.push('/store/bottomwear')}}
			className="rightlow zone shadow-1"><p className='texter shadow-1'>Bottomwear</p></div>
	</div>
	)
}

export default Main;