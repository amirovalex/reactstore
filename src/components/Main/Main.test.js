import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main.js';


it('expect to render component',() => {
	const mainComponent = shallow(<Main/>)
	expect(mainComponent.debug()).toMatchSnapshot();
})