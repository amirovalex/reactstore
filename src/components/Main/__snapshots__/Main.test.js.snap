// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render component 1`] = `
"<div className=\\"container\\">
  <div onClick={[Function: onClick]} className=\\"left shader zone tc\\">
    <p className=\\"glow texter shadow-1\\">
      Accessories
    </p>
  </div>
  <div onClick={[Function: onClick]} className=\\"righttop zone shadow-1\\">
    <p className=\\"texter shadow-1\\">
      Topwear
    </p>
  </div>
  <div onClick={[Function: onClick]} className=\\"rightlow zone shadow-1\\">
    <p className=\\"texter shadow-1\\">
      Bottomwear
    </p>
  </div>
</div>"
`;
